n2y/utils.py

```diff
@@ -93,6 +94,51 @@ def pandoc_write_or_log_errors(pandoc_ast, format, options):
             raise
 
 
+def yaml_to_meta_value(data):
+    """
+    Convert a Python object to a Pandoc metadata values, following the approach used by
+    `yamlToMetaValue` here:
+
+    https://github.com/jgm/pandoc/blob/main/src/Text/Pandoc/Readers/Metadata.hs.
+
+    Note that all scalars end up as strings.
+    """
+    if isinstance(data, str):
+        return MetaString(data)
+    elif isinstance(data, bool):
+        return MetaBool(data)
+    elif data is None:
+        return MetaString("")
+    elif isinstance(data, numbers.Number):
+        return MetaString(str(data))
+    elif isinstance(data, list):
+        return MetaList([yaml_to_meta_value(item) for item in data])
+    elif isinstance(data, dict):
+        return MetaMap({key: yaml_to_meta_value(value) for key, value in data.items()})
+
+
+def yaml_map_to_meta(data):
+    assert isinstance(data, dict)
+    assert all(isinstance(k, str) for k in data.keys())
+    return Meta({k: yaml_to_meta_value(v) for k, v in data.items()})
+
+
+def pandoc_format_to_file_extension(format):
+    # split on '-' or '+' to handle formats like 'markdown+raw_tex'
+    base_type = re.split(r'[-+]', format)[0]
```

Comment:
Do you not ever need to consider the post `+` or `-` extension type?

Reply:
I don't believe so, as I have yet to see an extension that so drastically changes the format that we'd need to adjust the file type to it.